### PR TITLE
2183 concent/forced download force get task results download

### LINKED ISSUE #2183: [Concent] UC2 / Forced Download - react to `ForceGetTaskResultDownload`
implement reaction to `ForceGetTaskResultDownload`.

actually this means that the results need to be downloaded from the Concent in this scenario instead of the Provider

### PR DESCRIPTION
**Pull request by shadeofblue**
closes #2183

### CODE REVIEW

File: golem/task/result/resultmanager.py [hunk 1]
```diff
@@ -36,13 +36,18 @@ def __init__(self, resource_manager):
...
+        return file_name, file_path
+
     # Using a temp path
     def pull_package(self, content_hash, task_id, subtask_id, key_or_secret,
                      success, error, async_=True, client_options=None, output_dir=None):
 
-        file_name = task_id + "." + subtask_id
-        file_path = self.resource_manager.storage.get_path(file_name, task_id)
-        output_dir = os.path.join(output_dir or os.path.dirname(file_path), subtask_id)
+        file_name, file_path = self.get_file_path_and_name(task_id, subtask_id)
```
Comment (mfranciszkiewicz) [inline golem/task/result/resultmanager.py]: The name suggests a different result order :)

File: golem/network/concent/received_handler.py [hunk 2]
```diff
@@ -310,3 +317,70 @@ def error(exc):
...
+            rct.task_id, rct.subtask_id)
+
+        task = self.task_server.task_manager.tasks.get(rct.task_id, None)
+        output_dir = task.tmp_dir if hasattr(task, 'tmp_dir') else None
+
+        def success(response):
+            logger.debug("Concent results download sucessful: %r, %s",
+                         msg.subtask_id, response)
+
+            extracted_package = result_manager.extract(
```
Comment (mfranciszkiewicz) [inline golem/network/concent/received_handler.py]: + some error handling

File: golem/core/printable_object.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    """
+    a simple mixin that provides a default implementation of the `__str__`
+    method that serializes the objects `__dict__` to the string for easier
+    human-readable represenation of simple, dictionary-like objects
+    """
+
+    def __str__(self):
+        return '%s <%s>' % (
+            self.__class__.__name__,
+            ', '.join(['%s: %s' % (k, v) for k, v in self.__dict__.items()])
```
Comment (jiivan) [inline golem/core/printable_object.py:14]: I suggest `__qualname__` and standard dict formatting.
```python

return '{} {}'.format(self.__class__.__qualname__, self.__dict__)
```

File: golem/network/concent/received_handler.py [hunk 4]
```diff
@@ -281,8 +283,13 @@ def on_ack_force_get_task_result(self, msg, **_):
 
     # pylint:enable=no-self-use
```
Comment (jiivan) [inline golem/network/concent/received_handler.py]: No t necessary

File: golem/network/concent/received_handler.py [hunk 5]
```diff
@@ -281,8 +283,13 @@ def on_ack_force_get_task_result(self, msg, **_):
...
 
+    @staticmethod
+    def _log_ftt_invalid(msg: message.base.Message):
+        logger.warning("File Transfer Token invalid in %r", msg)
```
Comment (jiivan) [inline golem/network/concent/received_handler.py:288]: This method  could be a function. It's related to message, not Messagehandler class.

File: golem/core/printable_object.py (hunk 3 above)
Comment (jiivan) [reply golem/core/printable_object.py:14]: or even `pprint.pformat(self.__dict__)`

**Review by jiivan — APPROVED** [review]
LGTM, small suggestions

File: golem/core/printable_object.py (hunk 3 above)
Comment (shadeofblue) [reply golem/core/printable_object.py:14]: @jiivan not quite... 

```
In [1]: class JivanPrintable:
   ...:     def __str__(self):
   ...:         import pprint
   ...:         return pprint.pformat(self.__dict__)
   ...:     

In [2]: from golem.core.printable_object import PrintableObject

In [3]: class A(PrintableObject):
   ...:     def __init__(self):
   ...:         self.dupa = 'dupa'
   ...:         

In [4]: class AA(PrintableObject):
   ...:     def __init__(self):
   ...:         self.a = A()
   ...:         

In [5]: class B(JivanPrintable):
   ...:     def __init__(self):
   ...:         self.dupa = 'dupa'
   ...:         

In [6]: class BB(JivanPrintable):
   ...:     def __init__(self):
   ...:         self.b = B()
   ...:         

In [7]: a = A()

In [8]: b = B()

In [9]: aa = AA()

In [10]: bb = BB()

In [11]: str(a)
Out[11]: 'A <dupa: dupa>'

In [12]: str(b)
Out[12]: "{'dupa': 'dupa'}"

In [13]: str(aa)
Out[13]: 'AA <a: A <dupa: dupa>>'

In [14]: str(bb)
Out[14]: "{'b': <__main__.B object at 0x7f16fe2da668>}"

In [15]: 
```

File: golem/network/concent/received_handler.py (hunk 5 above)
Comment (shadeofblue) [reply golem/network/concent/received_handler.py:288]: hmm, right, but in such a case, it should then be moved into the API of the appropriate `ForceGetTaskResult[Upload|Download]` classes ... I'll leave that as an excercise for some future update ... generally, this piece of code was earlier included as a part of the handler and just moved outside of the method for DRY reasons...

**Pull request merged**